woocommerce/woocommerce-android
# Issue/2710 Delete downloadable files

Fixes #2710 

**Description**
Adds ability to delete individual downloadable files for products.
The main changes are:
1. Update `ProductDownloadDetailsFragment` to handle the click on the option menu "Delete"
2. Update `ProductDownloadDetailsViewModel` to handle handle the file deletion, and trigger a confirmation dialog.
3. Add a test case for the file deletion.
4. Handled the case if a file's name is empty: https://github.com/woocommerce/woocommerce-android/pull/2742/commits/00340ee6dd28b186c33297384a306537b8076540
5. Updated the logic to show the `done` button in the files list if the settings were changed: https://github.com/woocommerce/woocommerce-android/pull/2742/commits/ffe7abaa41ae02ff17442f5c3f8c153a82ee68a3

**Screenshots**
![Screen Shot 2020-08-18 at 19 14 07](https://user-images.githubusercontent.com/1657201/90550756-435a1380-e188-11ea-8cfc-e2f242ea2af1.png)
![Screen Shot 2020-08-18 at 19 14 13](https://user-images.githubusercontent.com/1657201/90550781-4e14a880-e188-11ea-8020-e2e46764a191.png)
![delete-files](https://user-images.githubusercontent.com/1657201/90550804-566ce380-e188-11ea-96b4-fa6ee245c957.gif)

**Testing**
1. Make sure your product is downloadable and has some downloadable files added in Core.
2. Go to the product details, then click on "Downloadable files"
3. Click on the file title.
4. Click on the overflow button to see the option "Delete"
5. Click on Delete and confirm that the confirmation dialog is shown.
6. Click on Yes, and confirm that you are taken back to:
      a. The list of files if your product has more files.
      b. The product details if this is the last file.
